app/models/manageiq/providers/redhat/infra_manager/api_integration.rb

```diff
@@ -30,12 +31,19 @@ def connect(options = {})
     # Create the underlying connection according to the version of the oVirt API requested by
     # the caller:
     connect_method = "raw_connect_v#{version}".to_sym
-    connection = self.class.public_send(connect_method, server, port, path, username, password, service)
+    unless @connection.nil?
```

Comment:
There is a [`Connection#test`](http://www.rubydoc.info/gems/ovirt-engine-sdk/OvirtSDK4%2FConnection%3Atest) method, but only for version 4 of the API, and the connection you are handling here may be version 3 or version 4, so you can't use it directly.

In addition, I wonder how this would work. If I understand correctly you are using an instance attribute to store the reference to the connection. That means that we will have potentially as many connections as instances of the external management system type. I mean, each time you load the EMS from the database we will be potentially creating a new connection. When will those connections be closed? Connections need to be closed in order to avoid leaking resources in the oVirt/RHV server side (specially authentication sessions and HTTP connections). If we decide to implement this kind of connection optimization then I'd suggest to create singleton that contains a hash. The keys of that hash should be the identifiers of the EMSs, and the values the connections. We would also need a mechanism to close the those connections when they are no longer needed, at least when the Ruby process goes down.